File: python3/Fast and Slow Pointers/linked_list_loop_naive.py
from ds import ListNode

"""
Definition of ListNode:
class ListNode:
    def __init__(self, val=None, next=None):
        self.val = val
        self.next = next
"""


def linked_list_loop_naive(head: ListNode) -> bool:
    visited = set()  
    curr = head
    while curr:  
        # Cycle detected if the current node has already been visited.
        if curr in visited:
            return True  
        visited.add(curr)  
        curr = curr.next
    return False  

File: csharp/Fast and Slow Pointers/LinkedListLoopNaive.cs
using ds.ListNode;

/*
	Definition of ListNode:
	class ListNode
	{
		public ListNode(int val, ListNode next = null)
		{
			Val = val;
			Next = next;
		}

		public int Val { get; set; }

		public ListNode Next { get; set; }
	}
*/

public class Solution
{
	public bool LinkedListLoopNaive(ListNode head)
	{
		ISet<ListNode> visited = new HashSet<ListNode>();
		ListNode curr = head;

		while (curr != null)
		{
			//Cycle detected if the current node has already been visited.
			if (visited.Contains(curr))
				return true;

			visited.Add(curr);
			curr = curr.Next;
		}

		return false;
	}
}